### PR TITLE
make sure we clean up the ISTIO_OUTPUT chain

### PR DESCRIPTION
**Pull request by dhawton**
**Please provide a description of this PR:**

Running `pilot-agent istio-clean-iptables` would leave the ISTIO_OUTPUT chain in existance that was added by https://github.com/istio/istio/pull/50915.

Tests covered by https://github.com/istio/istio/pull/52860

Fixes https://github.com/istio/istio/issues/52835

cc: @howardjohn

### CODE REVIEW

File: tools/istio-clean-iptables/pkg/cmd/cleanup.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func removeOldChains(cfg *config.Config, ext dep.Dependencies, iptV *dep.Iptable
 	}
 
 	// Must be last, the others refer to it
-	chains = []string{constants.ISTIOREDIRECT, constants.ISTIOINREDIRECT}
+	chains = []string{constants.ISTIOREDIRECT, constants.ISTIOINREDIRECT, constants.ISTIOOUTPUT}
```
Comment (bleggett) [inline tools/istio-clean-iptables/pkg/cmd/cleanup.go:84]: Yeah this is one of those cases where we shouldn't be duplicating cleanup stuff across separate binaries, really.

There is no real benefit to it and it's just another thing to keep in sync.

**Review by bleggett — APPROVED** [review]

File: tools/istio-clean-iptables/pkg/cmd/cleanup.go (hunk 1 above)
Comment (dhawton) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go:84]: Agreed. I saw that PR but seems like more discussions are occurring so this should resolve the issue for now and we can iterate a better procedure.

Comment (leosarra) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go:84]: fyi I have made #52918 (which relies the work done in #50328) to have VMs use the new istio-iptables --cleanup-option.

**Comment by zhaohuabing** [comment]
@dhawton Could this be cherry-picked into 1.22 as well?

**Comment by deepakdeore2004** [comment]
> @dhawton Could this be cherry-picked into 1.22 as well?

my bad that i thought i used 1.22 rpm file but it was 1.23, this issue isnt present in 1.22

**Pull request merged**